src/connectors/impls/kafka/consumer.rs

```diff
@@ -37,18 +39,30 @@ const KAFKA_CONSUMER_META_KEY: &str = "kafka_consumer";
 
 #[derive(Deserialize, Debug, Clone)]
 #[serde(deny_unknown_fields)]
-pub struct Config {
+pub(crate) struct Config {
     /// consumer group id to register with
-    pub group_id: String,
+    group_id: String,
     /// List of topics to subscribe to
-    pub topics: Vec<String>,
+    topics: Vec<String>,
     /// List of bootstrap brokers
-    pub brokers: Vec<String>,
+    brokers: Vec<String>,
+
+    /// Whether or not to commit offsets of received messages immediately
+    /// or only upon received ack messages.
+    ///
+    /// If `auto_commit` is set to `false`, events emitted from this consumer
+    /// are considered transactional, soliciting an ack/fail message upon successful or failed handling downstream.
+    #[serde(default = "default_true")]
+    auto_commit: bool,
```

Comment:
I'm not sure it's a good idea to pull out more variables form `config` in here and have them definable in two places, that could quickly lead to confusion.

It also means that our defaults might not be in sync with the defaults from librdkafka leading to unexpected results - unless we make sure that we're always in sync which we will not.

Reply:
We overwrite user config with the settings coming from this and if we overwrite user setting, we issue a warning. We ensure the client working sensibly, by enforcing `enable.auto.commit` no matter what. It is meant to prevent people shooting themselves in the foot and allowing them to setup their kafka connector without the need to dive into librdkafka configs.

We might want to add a third mode where we do the current behaviour and send a commit for every single message to the group coordinator.

Another way we could work with the config is to simply look if people have enabled auto commit in their kafka config and store the offsets instead of sending them to the coordinator.